### ISSUE TITLE
ngrx/effects: SCRIPT1003 in IE 11

### ISSUE THREAD

**Issue by john-mhm** [ngrx/platform#682]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?
When Angular loads, I receive the following error:
SCRIPT1003: Expected ':' in vendor.bundle.js which points to the following line:
    return { memoized, reset };

## Expected behavior:
The site loads in IE 11.

## Minimal reproduction of the problem with instructions:
I was unable to reproduce. I tried to set up a new project with the angular cli (1.6.3) but that seems to create a project that does not work in IE 11 (different error, though). It did not seem to break in plunkr so I assume it's some intricacy with our project.

## Version of affected browser(s),operating system(s), npm, node and ngrx:
IE11, Win 10, npm 5.5.1, node 8.9.3, ngrx 4.0.2 and 4.1.1

## Other information:
This seems to be similar to #https://github.com/ngrx/platform/issues/171
I have attempted the following steps to resolve this:

- Removed third-party npm packages
- Removed code using ngrx/effects
- Removed ngrx/effects (site worked when removed)

This started as an Angular 2 project and we upgraded to 4.4.1 before installing ngrx 4.0.2. I've updated everything to latest (including to Angular 5/ngrx 4.1.1) based on the referenced issue above in case something was out of date to what it should be, but to no avail. I've tried the suggestions in the referenced issue with both Angular 4.4.1 and Angular 5. It seems to fail even removing all effects but still importing the EffectsModule which leads me to believe it's some peculiarity with packages installed.

**The current package.json looks like:**
```
{
  "name": "insight",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@agm/core": "^1.0.0-beta.2",
    "@angular/animations": "^5.1.2",
    "@angular/cdk": "^5.0.2",
    "@angular/common": "^5.1.2",
    "@angular/compiler": "^5.1.2",
    "@angular/core": "^5.1.2",
    "@angular/forms": "^5.1.2",
    "@angular/http": "^5.1.2",
    "@angular/material": "^5.0.2",
    "@angular/platform-browser": "^5.1.2",
    "@angular/platform-browser-dynamic": "^5.1.2",
    "@angular/router": "^5.1.2",
    "@angular/upgrade": "^5.1.2",
    "@ngrx/effects": "^4.1.1",
    "@ngrx/router-store": "^4.1.1",
    "@ngrx/store": "^4.1.1",
    "angular-2-dropdown-multiselect": "^1.6.3",
    "angular2-cookie": "^1.2.6",
    "angular2-material-datepicker": "^0.5.0",
    "core-js": "^2.5.3",
    "font-awesome": "^4.7.0",
    "lodash": "^4.17.4",
    "pnotify": "^3.2.1",
    "rxjs": "^5.5.6",
    "themify-icons": "^1.0.0",
    "zone.js": "^0.8.19"
  },
  "devDependencies": {
    "@angular/cli": "1.6.3",
    "@angular/compiler-cli": "^5.1.2",
    "@angular/language-service": "^5.1.2",
    "@types/googlemaps": "^3.30.3",
    "@types/jasmine": "2.8.3",
    "@types/lodash": "^4.14.91",
    "@types/node": "~8.5.2",
    "codelyzer": "~4.0.2",
    "jasmine-core": "~2.8.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "^2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.3.3",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "node-sass": "^4.7.2",
    "protractor": "~5.2.2",
    "ts-node": "~4.1.0",
    "tslint": "~5.8.0",
    "typescript": "~2.6.2"
  }
}
```

**The previous package.json before any changes:**
```
{
  "name": "insight",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@agm/core": "^1.0.0-beta.1",
    "@angular/animations": "^4.4.1",
    "@angular/cdk": "^2.0.0-beta.12",
    "@angular/common": "^4.4.1",
    "@angular/compiler": "^4.4.1",
    "@angular/core": "^4.4.1",
    "@angular/forms": "^4.4.1",
    "@angular/http": "^4.4.1",
    "@angular/material": "^2.0.0-beta.12",
    "@angular/platform-browser": "^4.4.1",
    "@angular/platform-browser-dynamic": "^4.4.1",
    "@angular/router": "^4.4.1",
    "@angular/upgrade": "^4.4.1",
    "@ngrx/core": "^1.2.0",
    "@ngrx/effects": "^4.0.2",
    "@ngrx/router-store": "^4.0.2",
    "@ngrx/store": "^4.0.2",
    "@types/googlemaps": "^3.26.20",
    "angular-2-dropdown-multiselect": "^1.6.3",
    "angular-font-awesome": "^2.0.2",
    "angular2-cookie": "^1.2.6",
    "angular2-material-datepicker": "^0.5.0",
    "core-js": "^2.5.1",
    "font-awesome": "^4.7.0",
    "lodash": "^4.17.4",
    "pnotify": "^3.2.1",
    "rxjs": "^5.1.0",
    "themify-icons": "^1.0.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "1.1.1",
    "@angular/compiler-cli": "^4.4.1",
    "@angular/language-service": "^4.4.1",
    "@types/jasmine": "2.5.45",
    "@types/lodash": "^4.14.68",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "^1.7.1",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "node-sass": "^4.6.0",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
}
```

**Comment by brandonroberts** [comment]
Please provide a reproduction of this issue.

**Comment by john-mhm** [comment]
I attempted to reproduce but couldn't and after rebuilding a new project and slowly inserting pieces in, have isolated the module it's occurring in (not ngrx), so I'm closing this up. Thanks for taking a look!

**Comment by john-mhm** [comment]
@brandonroberts I did eventually run into this problem again, and the fix was entirely the same as #711. I just stumbled across it when trying to figure it out. Guess I need to watch my imports. And the resolution in #711 fixed the issue.
